llama-index-core/llama_index/core/schema.py

```diff
@@ -526,6 +526,15 @@ def resolve_image(self) -> ImageType:
         else:
             raise ValueError("No image found in node.")
 
+    @property
+    def hash(self) -> str:
+        """Get hash of node."""
+        # doc identity depends on if image, image_path, or image_url is set
+        image_str = self.image or "None"
+        image_path_str = self.image_path or "None"
+        image_url_str = self.image_url or "None"
+        return f"{image_str}-{image_path_str}-{image_url_str}"
```

Comment:
The `image_str` could be huge here -- maybe better to actually hash it?

Reply:
Technically image nodes also have a `.text` component, we should include that as well

Reply:
good catch i meant to hash